### PR TITLE
Welcome home ScopedInstance

### PR DESCRIPTION
**Pull request by HaoK**
Moved from DI

### CODE REVIEW

**Comment by HaoK** [comment]
Also removes configuration from AddMvc now that Authorization no longer needs it

**Comment by dougbu** [comment]
please copy over the `ScopedInstanceTest` class

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -15,16 +15,10 @@ namespace Microsoft.Framework.DependencyInjection
...
     {
-        public static IServiceCollection AddMvc([NotNull] this IServiceCollection services)
-        {
-            return AddMvc(services, configuration: null);
-        }
-
         public static IServiceCollection AddMvc(
-            [NotNull] this IServiceCollection services,
-            IConfiguration configuration)
+            [NotNull] this IServiceCollection services)
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: nit: no need to wrap here

**Comment by davidfowl** [comment]
:shipit:

**Comment by HaoK** [comment]
Moved test over and fixed wrap

File: test/Microsoft.AspNet.Mvc.Core.Test/ScopedInstanceTests.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ScopedInstanceTests
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Core.Test/ScopedInstanceTests.cs]: rename class and file to `...Test` per guidelines

**Comment by dougbu** [comment]
:shipit:

**Comment by HaoK** [comment]
Updated PR to include fixing all the startup's for all the functional tests :/

File: test/WebSites/ActionConstraintsWebSite/Startup.cs [hunk 3]
```diff
@@ -11,11 +11,9 @@ public class Startup
     {
         public void Configure(IApplicationBuilder app)
         {
-            var configuration = app.GetTestConfiguration();
```
Comment (dougbu) [inline test/WebSites/ActionConstraintsWebSite/Startup.cs]: no.  do not remove this line.  the method configures a required middleware for the functional tests.  we can rename it and make it `void` at some point.  but it must remain.

**Comment by dougbu** [comment]
:watch:

**Comment by HaoK** [comment]
So i removed them all and I see that it causes a few failures, but its only 4...

TagHelpers, MvcTagHelpers, ModelBinding (culture replacement), how I about I just leave it for these 4 websites instead of everywhere?

**Comment by dougbu** [comment]
we want to consistently run the functional tests with this middleware in place.  that's why the setup is in a central method and not the individual web site's `Startup()`.  so, no.

**Comment by HaoK** [comment]
That's fine, I'm going to revert the change to removing configuration from AddMvc and file a issue for MVC to remove it as a separate PR, this one is just for ScopedInstance

**Comment by HaoK** [comment]
Updated PR to decouple AddMvc changes for now

**Comment by dougbu** [comment]
@HaoK why did you revert all the `Startup` changes instead of just the `app.GetTestConfiguration()` calls?

**Comment by HaoK** [comment]
CI build is broken, I just wanted to move IScopedInstance to MVC, I was doing some additional cleanup, but its become too noisy, I want to push a clean PR now

**Comment by HaoK** [comment]
I backed out the breaking change to Mvc as that isn't something I need now

**Comment by dougbu** [comment]
shoot, wish you'd held off on pushing the DI change
:shipit:

**Comment by HaoK** [comment]
f3a9ce1ec176bcaf9170e9c984fd5fd94c9ec851

**Pull request closed without merging**